src/main/java/io/boomerang/eventing/nats/jetstream/PullBasedSubReceiver.java

```diff
@@ -100,7 +100,6 @@ public void run() {
                 .forEachRemaining(consumer);
           } catch (IllegalStateException e) {
             logger.error("An exception was raised when pulling new messages from the consumer!", e);
-            return;
```

Comment:
Hey @AndreiOla, this is a potential solution, but I'm not certain how this would behave if you programmatically stop the receiver by invoking `SubReceiver.unsubscribe()`. If I remember correctly, the `IllegalStateException` is raised by `jnats` is we forcefully stop the consumer while its active (consuming messages), but the JetStream connection is closed. Please look into this and ensure it doesn't brake the unsubscribe functionality.